modules/type/minion/minion.gd

```diff
@@ -0,0 +1,66 @@
+extends Node
+
+
+#region Internal Functions
+func _ready() -> void:
+	Modules.register(&"Minion", [&"Type", &"Attack", &"Health"], func() -> void:
+		# Load module.
+		Modules.register_hooks(&"Minion", self.handler)
+		
+		TypeModule.register_type(&"Minion", true)
+		AttackModule.register(&"Minion")
+		HealthModule.register(&"Minion")
+	, func() -> void:
+		# Unload module. No need to unregister hooks.
+		TypeModule.unregister_type(&"Minion")
+	)
+#endregion
+
+
+#region Public Functions
+func handler(what: Modules.Hook, info: Array) -> bool:
+	if what == Modules.Hook.ACCEPT_PACKET:
+		return await accept_packet_hook.callv(info)
+	
+	return true
+
+
+func is_minion(card: Card) -> bool:
+	return TypeModule.is_card_type(card, &"Minion")
+
+
+func accept_play_packet(
+	player: Player,
+	sender_peer_id: int,
+	
+	location: StringName,
+	location_index: int,
+	board_index: int,
+	position: Vector3i,
+) -> bool:
+	var card: Card = Card.get_from_index(player, location, location_index)
+	
+	if not is_minion(card):
+		return true
+	
+	if card.abilities.has(&"Battlecry"):
+		card.trigger_ability(&"Battlecry", false)
+		await card._wait_for_ability(&"Battlecry")
+		
+		if card.refunded:
+			return false
+	
+	# Summon after ability for refunding.
+	# Kinda gross...
+	card.player.summon_card(card, board_index, false)
```

Comment:
I think this is gross since we are summoning the card after triggering the ability, which might be confusing to card creators if they don't know the reasoning behind it. This should be documented somewhere, but in the meantime, rephrase this comment with the reasoning above.

Also we should add `Card.summon` so we can do this instead:
```gdscript
card.summon(board_index, false)
```